core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1447,13 +1454,33 @@ class ConnectionQuotas(config: KafkaConfig, time: Time, metrics: Metrics) extend
       }
     }
 
+    def removeSensors(): Unit = {
```

Comment:
nit: What about naming this `close`? It feels a bit more natural to call `close` when we don't need the object anymore and it would be more aligned with the `close` methods that we already have in this file. For instance, `ConnectionQuotas#close` which also cleanup the metrics.

BTW, not related to this PR but it seems that we don't close the `ListenerConnectionQuota` when the `ConnectionQuotas` is closed. I suppose that we leave metrics around. Is it the case?

Reply:
agreed, `close` will be consistent. 

About not closing ListenerConnectionQuota on ConnectionQuotas.close(): this is on broker shutdown path where KafkaServer calls metrics.close() on shutdown as well (and metrics are owned by KafkaServer), so I don't think we are leaking anything. But I think it's better to remove listener sensors on ConnectionQuotas.close() anyways, so I will make that change.